esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -1102,44 +1076,32 @@ unsigned WsWuInfo::getWorkunitThorLogInfo(IArrayOf<IEspECLHelpFile>& helpers, IE
                 else
                     fileType.appendf("%s%d", File_ThorLog, countThorLog);
 
-                helpersCount++;
-                if (flags & WUINFO_IncludeHelpers)
-                {
-                    Owned<IEspECLHelpFile> h= createECLHelpFile("","");
-                    h->setName(logName.str());
-                    h->setDescription(processName.str());
-                    h->setType(fileType.str());
-                    if (version >= 1.43)
-                    {
-                        offset_t fileSize;
-                        if (getFileSize(logName.str(), NULL, fileSize))
-                            h->setFileSize(fileSize);
-                    }
-                    helpers.append(*h.getLink());
-                }
-
-                if (version < 1.38)
-                    continue;
-
-                const char* pStr = logName.str();
-                const char* ppStr = strstr(pStr, "/thormaster.");
-                if (!ppStr)
+                Owned<IEspECLHelpFile> h = createECLHelpFile();
+                const char *logName = logInfo.getLogName();
+                h->setName(logName);
+                h->setDescription(logInfo.getProcessName());
+                h->setType(fileType.str());
+                if (version >= 1.43)
                 {
-                    IWARNLOG("Invalid thorlog entry in workunit xml: %s", logName.str());
-                    continue;
+                    offset_t fileSize;
+                    if (getFileSize(logName, nullptr, fileSize))
+                        h->setFileSize(fileSize);
                 }
+                helpers.append(*h.getLink());
+            }
 
-                ppStr += 12;
-                StringBuffer logDate(ppStr);
-                logDate.setLength(10);
+            if (version < 1.38)
+                continue;
 
-                Owned<IEspThorLogInfo> thorLog = createThorLogInfo("","");
-                thorLog->setProcessName(processName.str());
-                thorLog->setClusterGroup(groupName.str());
-                thorLog->setLogDate(logDate.str());
-                thorLog->setNumberSlaves(numberOfSlaveLogs);
-                thorLogList.append(*thorLog.getLink());
+            Owned<IEspThorLogInfo> thorLog = createThorLogInfo();
+            thorLog->setProcessName(logInfo.getProcessName());
+            if (version < 1.78)
+            {
+                thorLog->setClusterGroup(logInfo.getGroupName());
+                thorLog->setLogDate(logInfo.getLogDate());
```

Comment:
The logDate is correct. The groupName may not be correct for imported WU. I may skip that if Debug Value "imported" is set. Please comment.

Reply:
I think what's confusing is that the getWUThorLogInfoLW vs getWUThorLogInfo conditional code - there should be 1 interface method, it can fill-in/deduce more than is required by newer clients.

This code may be fine though, simply using some fields that may be present from the IConstWUThorLogInfo instances.